cddl/concise-mid-tag.cddl

```diff
@@ -1,62 +1,123 @@
+;
+impl-id = bytes .size 32
+tagged-impl-id = #6.47115(impl-id)
+
+;
+; From draft-ietf-rats-eat
+;
+ueid = bstr .size 33
+tagged-ueid = #6.48000(ueid)
+
+module-version = {
+  0 => text .default '0.0.0', ; version
+  1 => $version-scheme,       ; version-scheme
+}
+
+; version-scheme defined in CoSWID
+
+module-vendor = text
+
+linked-tags = 106
 
 linked-tags-entry = {
-  tag-id => $tag-id-type,
-  tag-rel => $tag-rel-type,
-  ? tag-thumbprint => hash-entry, ; digest of the CBOR encoding of the concise-tag
-  ? tag-issuer => tag-issuer-type,
+  0 => $tag-id-type,
+  1 => $tag-rel-type,
+  ; tag thumbprint, i.e., digest of the CBOR encoding of the concise-tag
+  ? 2 => hash-entry,
+  ; verification key associated with the tag issuer
+  ? 3 => tag-issuer-type,
 }
-tag-rel=121
-tag-thumbprint = 199 ; NOTE(tho) - randomly assigned number: check this for clashes
-tag-issuer = 200 ; NOTE(tho) - randomly assigned number: check this for clashes
 
-; Bob is saying I expect this tag to be provided (signed by) by Alice
+; Bob is saying: I expect this tag to be provided (signed by) by Alice.
+; The COSE keyset only contains raw / certified public keys.
 tag-issuer-type /= COSE_KeySet
 
-$tag-rel-type /= parent ; a tag that introduces additional element-names
-$tag-rel-type /= supplemental ; a tag that augments existing element-names with additional values
-$tag-rel-type /= supersedes ; a tag that replaces existing element-names and values
-$tag-rel-type /= -256..64436 / text
-parent = 0
-supplemental = 1
-supersedes = 2
+$tag-rel-type /= comid-includes
+$tag-rel-type /= comid-or-includes
+$tag-rel-type /= comid-supplements
+$tag-rel-type /= comid-updates
+$tag-rel-type /= comid-replaces
+$tag-rel-type /= comid-patches
+
+; The current module has additional module(s) identified by the linked tag(s).
+; The union of all modules describes an acceptable device.
+comid-includes = 0
+; The current module has additional module(s) identified by the linked tag(s)
+; that describe alternative acceptable modules. At least one of the
+; alternatives is included to describe an acceptable device.
+comid-or-includes = 1
+; The current module has additional claims that augment the claim set of the
+; target module. The current tag module information must match the target
+; module for the claims to be combined under the target.
+comid-supplements = 2
+; The current module contains claims that fully replace the target module
+; claims. This is applied when the module is updated or changes in some
+; fashion.
+comid-updates = 3
+; The current tag contains corrections to a previously issued tag. This tag
+; replaces the target tag. The tag-id remains unchanged. The tag-version is
+; incremented.
+comid-replaces = 4
+; The current module contains claims that partially replace the target module
+; claims. This is applied when the module is updated or changes in some
+; fashion.  The target module contains both valid and replaced claims.
+comid-patches = 5
```

Comment:
Do we want to define 'patches' yet? Hoping to avoid this complexity if possible.

Reply:
It is the tool we were planning to use in order to model FW patch releases.